### ISSUE TITLE
asciiquarium segfaults with brewed perl

### ISSUE THREAD

**Issue by apjanke** [Homebrew/homebrew-core#4936]
`asciiquarium` is segfaulting for me on OS X 10.11, both when poured from bottle and installed from source.

```
[~]
$ brew install asciiquarium
==> Downloading https://homebrew.bintray.com/bottles/asciiquarium-1.1.el_capitan.bottle.ta
Already downloaded: /Users/janke/Library/Caches/Homebrew/asciiquarium-1.1.el_capitan.bottle.tar.gz
==> Pouring asciiquarium-1.1.el_capitan.bottle.tar.gz
🍺  /usr/local/Cellar/asciiquarium/1.1: 16 files, 366.7K
[~]
$ asciiquarium
[1]    40607 segmentation fault  asciiquarium
[✘ ~]
$
```

Updating the `Curses` resource to the latest did not fix.

Is this still working for people on earlier versions of OS X? Should we put a `MaximumMacOS` requirement on this or boneyard it?

**Comment by zmwangx** [comment]
The interesting thing is it doesn't segfault on CI: https://bot.brew.sh/view/Testing/job/Homebrew%20Testing/1087/version=el_capitan/console

**Comment by apjanke** [comment]
Might be something to do with how the CI is effectively running headless and not attached to a real terminal (I'm not sure what the right term for it is on OS X). Doing `brew test-bot asciiquarium` locally fails for me. Maybe it's doing some test on the terminal it or its session is attached to.

```
$ brew test-bot asciiquarium
==> brew doctor
...
==> brew test asciiquarium --verbose
==> FAILED
Testing asciiquarium
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20160918-44438-w4xuvv.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Taps/homebrew/homebrew-core/Formula/asciiquarium.rb --verbose
Error: asciiquarium: failed
Expected /\x1B\[\?10/ to match "".
==> brew uninstall --force asciiquarium
[✘ ~]
$
```

Simply doing `asciiquarium | cat` to get its `stdout` to not be attached to a terminal still segfaults for me. Might be something more at the session level it's doing.

**Comment by apjanke** [comment]
Hmm. The `asciiquarium` program itself is a single Perl script and it doesn't seem to be doing anything fancy, so it's unlikely a segfault could be fixed at that level. I'd guess it's more likely something in the `Curses` module. Bummer.

**Comment by ilovezfs** [comment]
I'm not able to reproduce this.

**Comment by zmwangx** [comment]
I'm totally able to reproduce this.

**Comment by ilovezfs** [comment]
<img width="1242" alt="screen shot 2016-09-17 at 11 47 05 pm" src="https://cloud.githubusercontent.com/assets/5268928/18613505/3af7dc3c-7d31-11e6-94a4-a271ad225434.png">

**Comment by ilovezfs** [comment]
It occurs with brew perl not system perl.

**Comment by apjanke** [comment]
What's your `brew config`, @ilovezfs?

Mine:

```
$ brew config
HOMEBREW_VERSION: 0.9.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7dd7e81dd101e4ac4bc67bb80663ee6a0076f5d9
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d67912f755d8a1c4439bc3aec825ea87424d620a
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit skylake
Homebrew Ruby: 2.0.0-p648
Clang: 7.3 build 703
Git: 2.10.0 => /usr/local/bin/git
Perl: /usr/local/bin/perl => /usr/local/Cellar/perl/5.24.0_1/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.3.1/bin/ruby
Java: 1.8.0_91
OS X: 10.11.6-x86_64
Xcode: 7.3.1 => /Applications/Xcode-7.3.1.app/Contents/Developer
CLT: 7.3.1.0.1.1461711523
X11: 2.7.9 => /opt/X11
[~]
```

I'm on a 5K iMac.

**Comment by apjanke** [comment]
Aha!

Yep. After a `brew unlink perl`, `asciiquarium` works for me too.

**Comment by zmwangx** [comment]
Let the wrapper run the libexec script with `/usr/bin/perl` I guess.

**Comment by ilovezfs** [comment]
or make it depend on brew perl or a perl requirement

**Comment by zmwangx** [comment]
> make it depend on brew perl

That's the recipe for failure.

**Comment by ilovezfs** [comment]
Incorrect. Add depends_on "perl" and rebuild.

**Comment by zmwangx** [comment]
> Add depends_on "perl" and rebuild.

Okay it does work, but too heavy-handed it seems.

**Comment by ilovezfs** [comment]
whatever works :)

But clearly the `env perl` in the right now is no good.

**Comment by zmwangx** [comment]
By the way `/usr/bin/perl` hasn't been upgraded for a while (including in Sierra). Another piece of abandonware?

**Comment by apjanke** [comment]
Switching it to build/run with the system `perl` fixed it for me. PR at #4966.

```
        system "/usr/bin/perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
```

**Comment by mafrosis** [comment]
Segfault here too folks

**Comment by ilovezfs** [comment]
add `depends_on "perl"` to the formula and then rebuild it from source `--build-from-source`

**Comment by zmwangx** [comment]
Fix in #6148.

**Comment by zmwangx** [comment]
Shipped #6148.
